### ISSUE TITLE
⌨ `Accordion` contain invalid heading structure

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#15011]
### ⌨ Accessibility and Expected Behaviour

Our `Accordion` component contains both a `h3` and `h*` (* depends on the `Trigger` `as` prop).

### WCAG 2.1 Criteria

- https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1#qr-content-structure-separation-programmatic

### 📋 Steps to Reproduce

1. Open our accordion component in storybook
2. Confirm that you see an `h3` and `h2` in the trigger

### 🕵️ Details

The markup for our trigger is:

- h3
  - button
    - h2

This has sevral problems with it. 

1. Heading ranks are out of order
2. You cannot put a heading inside of another heading

### 📸 Screenshot

<img width="565" height="187" alt="Image" src="https://github.com/user-attachments/assets/f487921a-9de0-4249-afcc-ef98649b5625" />

### 🙋‍♀️ Proposed Solution

By default, the `Accordion.Header` is a `h3`. You can provide the `asChild` prop to set the appropriate heading rank but, we do not.

I'm not sure what to do because the design requires we do some odd invalid HTML. Agian, I beleive we need to take a look at our accordions and simplify them significantly.

We have too much going on in the trigger element causing many accessibility issues and a very brittle component.

In the interim, I think maybe just making it a div? This will still be invlaid HTML but, it atleast will solve one major issue.

### ✅ Acceptance Criteria

- [ ] `Accordion.Trigger` only contains a single heading
- [ ] That heading is still the appropriate heading rank
